app/DynamicsAdapter/DynamicsAdapter.Web/PersonSearch/SearchResultService.cs

```diff
@@ -563,10 +565,19 @@ private async Task<bool> UploadEmails()
             }
             catch (Exception ex)
             {
-                _logger.LogError(ex.Message);
+                LogException(ex);
                 return false;
             }
         }
 
+        private void LogException(Exception ex)
```

Comment:
I would be better to make this one log message with properties.  This will log multiple statements for one error.  This also doesn't provide context what the error is from.  I was trying to find an example. https://blog.rsuter.com/logging-with-ilogger-recommendations-and-best-practices/

not perfect...

```
private void LogException(Exception ex)
{
  var webRequestException = ex as Simple.OData.Client.WebRequestException;
  if (webRequestException != null) 
  {
    var properties = new Dictionary<string, object>();
    properties.Add("AbsoluteUri", webRequestException.RequestUri?.AbsoluteUri);
    properties.Add("Response", webRequestException.Response);
    using logger.BeginScope(properties);
    log.LogError(ex);
  }
  else
  {
    log.LogError(ex);
  }
}

```

Reply:
The majority of our "most valuable" service logs in splunk indeed have properties. I have had to bifurcate my reports by data available by property, which are usually on the request level, and ones targeting "plain" log lines, which don't have additional context, such as service name etc. Here is an example of what I see on the Splunk side for most "enriched" logs: 
![image](https://user-images.githubusercontent.com/49213216/120897377-af541e00-c5da-11eb-9a09-4a82344719fa.png)

Reply:
right. Code changed. So, for WebRequestException (which is from SimpleOData), there will be two more properties: AbsoluteUri and Response

Reply:
Perfect! Thanks Peggy :)